File: back_end/120/lesson_4/easy_2/7.rb
# Explain what the @@cats_count variable does and how it works. What code would you need to write to test your theory?

class Cat
  @@cats_count = 0

  def initialize(type)
    @type = type
    @age  = 0
    @@cats_count += 1
    puts @@cats_count
  end

  def self.cats_count
    @@cats_count
  end
end

cat1 = Cat.new('tabby')
cat2 = Cat.new('siamese')

puts Cat.cats_count
# 1. @@cats_count is a class variable that keeps track of the number of 'cat' objects that belong to the class
# 2. Everytime an object is initialized, '@@cats_count' is incremented by 1. 

File: exercises/js_fundamentals/medium_1/7.js
function invoiceTotal(...args) {
  let result = 0

  for (let i = 0; i < args.length; i += 1) {
    result += args[i];
  }

  return result;
}

console.log(invoiceTotal(20, 30, 40, 50));          // works as expected
console.log(invoiceTotal(20, 30, 40, 50, 40, 40));  // does not work; how can you make it work?
